azurerm/tags.go

```diff
@@ -88,7 +88,10 @@ func flattenAndSetTags(d *schema.ResourceData, tagsMap map[string]*string) {
 	output := make(map[string]interface{}, len(tagsMap))
 
 	for i, v := range tagsMap {
-		output[i] = *v
+		// Filter out $type from tags object to avoid unexpected change on plan.
+		if i != "$type" {
+			output[i] = *v
+		}
```

Comment:
I think it's best to filter this on the individual resource where this needs to be filtered out (as people may have a legitimate use for the `$type` tag) - can we instead filter this out on the resource before calling this function? e.g.

```
tagsToOutput := make([]string, 0)
for i, v := range resp.Tags {
  tag = *v
  // Filter out $type from tags object to avoid unexpected change on plan.
  if tag != "$type" {
    tagsToOutput[i] = tag
  }
}
```

in addition - can we add some validation to ensure that the `$type` tag can't be added on that resource? We can do this using a `ValidateFunc` on the `tags` schema item for this resource :)

Reply:
What about the webhookAction?

Reply:
I've updated the code and tests.